src/util/LazyJsonParser.cpp

```diff
@@ -29,6 +29,20 @@ cppcoro::generator<nlohmann::json> LazyJsonParser::parse(
   }
 }
 
+// ____________________________________________________________________________
+cppcoro::generator<nlohmann::json> LazyJsonParser::parse(
+    cppcoro::generator<std::span<std::byte>> partialJson,
+    std::vector<std::string> arrayPath) {
+  return parse(
+      [](cppcoro::generator<std::span<std::byte>> partialJson)
+          -> cppcoro::generator<std::string> {
+        for (const auto& bytes : partialJson) {
+          co_yield reinterpret_cast<const char*>(bytes.data());
+        }
```

Comment:
This copies a string at least once more than necessary:

* once from the byte generator to the `string` generator
* then your lazy parser anyway has to copy, because you append to the previous input etc.
If that is correct what I am saying, then you can return a `generator<string_view> ` here and also consume `generator<string_view>` in your `parse` method.

If I am mistaken, and the lazy parser can always efficiently move from the passed in strings, then this is not an issue.